framework/src/media/utils/MediaUtils.cpp

```diff
@@ -68,10 +72,227 @@ audio_type_t getAudioTypeFromPath(std::string datapath)
 		medvdbg("audio type : unknown\n");
 		return AUDIO_TYPE_INVALID;
 	}
 }
+
+void mp3_header_parsing(unsigned char * header, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat)
+{
+	*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+	
+	unsigned char bit;
+	unsigned int MPEG_Version;	
+	bit = header[1];
+	bit >>= 3;
+	bit &= (unsigned char)3;
+
+	if (bit == 0) {
+		MPEG_Version = 2;
+	}
+	else if (bit == 2) {
+		MPEG_Version = 1;
+	}
+	else if (bit == 3) {
+		MPEG_Version = 0;
+	}
+	bit = header[2];
+	bit >>= 2;
+	bit &= 3;
+	
+	if (bit == 0) {
+		*sample_rate = 44100;
+	}
+	else if (bit == 1) {
+		*sample_rate = 48000;
+	}
+	else if (bit == 2) {
+		*sample_rate = 32000;
+	}
+	else{
+		//Reserved
+	}
+	*sample_rate >>= MPEG_Version;
+	bit = header[3];
+	bit >>= 6;
+	if (bit <= 2) {
+		*channel = 2;
+	}
+	else{
+		*channel = 1;
+	}
+
+	return;
+}
+
+void aac_header_parsing(unsigned char * header, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat)
+{
+	unsigned char bit;
+	
+	*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+
+	bit = header[2];
+	bit >>= 2;
+	bit &= 15;
+
+	switch (bit) {
+	case 0 :
+		*sample_rate = 96000;
+		break;
+	case 1 :
+		*sample_rate = 88200;
+		break;
+	case 2 :
+		*sample_rate = 64000;
+		break;
+	case 3 :
+		*sample_rate = 48000;
+		break;
+	case 4 :
+		*sample_rate = 44100;
+		break;
+	case 5 :
+		*sample_rate = 32000;
+		break;
+	case 6 :
+		*sample_rate = 24000;
+		break;
+	case 7 :
+		*sample_rate = 22050;
+		break;
+	case 8 :
+		*sample_rate = 16000;
+		break;
+	case 9 :
+		*sample_rate = 12000;
+		break;
+	case 10 :
+		*sample_rate = 11025;
+		break;
+	case 11 :
+		*sample_rate = 8000;
+		break;
+	case 12 :
+		*sample_rate = 7350;
+		break;
+	default :
+		//Reserved
+		break;
+	}
+
+	bit = header[2];
+	bit <<= 2;
+	bit |= (header[3]>>6);
+	bit &= 7;
+
+	switch (bit) {
+	case 0 :
+		//Defined in AOT Specifc Config
+		break;
+	case 1 :
+		*channel = 1;
+		break;
+	case 2 :
+		*channel = 2;
+		break;
+	case 3 :
+		*channel = 3;
+		break;
+	case 4 :
+		*channel = 4;
+		break;
+	case 5 :
+		*channel = 5;
+		break;
+	case 6 :
+		*channel = 6;
+		break;
+	case 7 :
+		*channel = 8;
+		break;
+	default :
+		//Reserved
+		break;
+	}
+	return ;
+}
+
+void wave_header_parsing(unsigned char * header, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat) //header 44byte
+{
+	*channel = header[22];
+
+	*sample_rate <<= 8;
+	*sample_rate |= header[24];
+	
+	unsigned short BitPerSample = 0;
+
+	BitPerSample |= header[35];
+	BitPerSample <<= 8;
+	BitPerSample |= header[34];
+
+	switch (BitPerSample) {
+	case 8 :
+		*pcmFormat = AUDIO_FORMAT_TYPE_S8;
+		break;
+	case 16 :
+		*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+		break;
+	case 32 :
+		*pcmFormat = AUDIO_FORMAT_TYPE_S32_LE;
+		break;
+	default :
+		break;	
+	}
+
+	return;
+}
+
+bool header_parsing(FILE *fp, audio_type_t AudioType, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat)
```

Comment:
You did very well, bool type is necessary to check getting format was succeeded or not.
F/W should follow below,
```
if (!utils::header_parsing) {
    return PLAYER_ERROR_NOT_SUPPORTED_FILE;
}
```
To do this, all header_parsing should changed to bool type, and we have to check return values.
Please change your logic refer to my comment.